src/tests/test_utils/common_test_utils/src/data_utils.cpp

```diff
@@ -202,58 +206,131 @@ size_t byte_size(const InferenceEngine::TensorDesc &tdesc) {
             break;
-        case Precision::U16:
-            fill_data_const<Precision::U16>(raw_data_ptr, raw_data_size, val);
+        case Type_t::i16:
+        case Type_t::u16:
+        case Type_t::f16:
+        case Type_t::bf16:
+            copy_7D<uint16_t>(src_ptr, src_strides, dst_ptr, dst_strides, dst_dims);
             break;
-        case Precision::I16:
-            fill_data_const<Precision::I16>(raw_data_ptr, raw_data_size, val);
+        case Type_t::u8:
+        case Type_t::i8:
+            copy_7D<uint8_t>(src_ptr, src_strides, dst_ptr, dst_strides, dst_dims);
             break;
         default:
-            IE_THROW() << "Unsupported precision by fill_data_const() function";
+            OPENVINO_THROW("Unsupported precision by fill_data_with_broadcast function");
     }
 }
 
-void fill_data_const(InferenceEngine::Blob::Ptr& blob, float val) {
-    fill_data_const(blob, std::vector<float> {val});
+template<ov::element::Type_t SRC_E, ov::element::Type_t DST_E>
+void copy_with_convert(ov::Tensor& src_tensor, ov::Tensor& dst_tensor) {
+    using SRC_TYPE = typename ov::fundamental_type_for<SRC_E>;
+    using DST_TYPE = typename ov::fundamental_type_for<DST_E>;
+
+    auto src_ptr = src_tensor.data<SRC_TYPE>();
+    auto src_size = src_tensor.get_size();
+
+    auto dst_ptr = dst_tensor.data<DST_TYPE>();
+
+    std::copy(src_ptr, src_ptr + src_size, dst_ptr);
+}
+
+ov::Tensor make_with_precision_convert(ov::Tensor& tensor, ov::element::Type_t prc) {
+    OPENVINO_ASSERT(tensor.get_element_type() == ov::element::Type_t::f32);
+    ov::Tensor new_tensor(prc, tensor.get_shape());
+
+#define CASE(_PRC) case ov::element::Type_t::_PRC: \
+        copy_with_convert<ov::element::Type_t::f32, ov::element::Type_t::_PRC> (tensor, new_tensor); break
+    switch (prc) {
+        CASE(f32); CASE(f16); CASE(i64); CASE(u64); CASE(i32); CASE(u32); CASE(i16); CASE(u16); CASE(i8); CASE(u8);
+        default: OPENVINO_THROW("Unsupported precision case");
+    }
+#undef CASE
+
+    return new_tensor;
+}
+
+void fill_data_with_broadcast(ov::Tensor& tensor, size_t axis, std::vector<float> values) {
+    ov::Shape value_dims(tensor.get_shape().size() - axis, 1);
+    value_dims.front() = values.size();
+    auto prc = tensor.get_element_type();
+
+    ov::Tensor values_tensor;
+    values_tensor = ov::Tensor(ov::element::Type_t::f32, value_dims, values.data());
+
+    if (prc != ov::element::Type_t::f32) {
+        values_tensor = make_with_precision_convert(values_tensor, prc);
+    }
+
+    fill_data_with_broadcast(tensor, values_tensor);
+}
+
+ov::Tensor make_reshape_view(const ov::Tensor &tensor, std::vector<size_t> new_shape) {
+    auto new_size = std::accumulate(new_shape.begin(), new_shape.end(), 1, std::multiplies<size_t>());
+    OPENVINO_ASSERT(new_size == tensor.get_size());
+
+    return ov::Tensor(tensor.get_element_type(), new_shape, tensor.data());
```

Comment:
Why do you need this helper?
I think you can create a tensor without this additional function:
```
ov::Shape new_shape;
OPENVINO_ASSERT(ov::shape_size(new_shape) == tensor.size());
auto new_tensor =  ov::Tensor(tensor.get_element_type(), new_shape, tensor.data());
```
Also you can use reshape instead of creation new tensor

Reply:
done